sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/TextLineNumber.java

```diff
@@ -0,0 +1,460 @@
+    Insets insets = getInsets();
+    int availableWidth = getSize().width - insets.left - insets.right;
+
+    // Determine the rows to draw within the clipped bounds.
+
+    Rectangle clip = g.getClipBounds();
+    int rowStartOffset = component.viewToModel(new Point(0, clip.y));
+    int endOffset = component.viewToModel(new Point(0, clip.y + clip.height));
+
+    while (rowStartOffset <= endOffset) {
+      try {
+        if (isCurrentLine(rowStartOffset))
+          g.setColor(getCurrentLineForeground());
+        else
+          g.setColor(getForeground());
+
+        // Get the line number as a string and then determine the
+        // "X" and "Y" offsets for drawing the string.
+
+        String lineNumber = getTextLineNumber(rowStartOffset);
+        int stringWidth = fontMetrics.stringWidth(lineNumber);
+        int x = getOffsetX(availableWidth, stringWidth) + insets.left;
+        int y = getOffsetY(rowStartOffset, fontMetrics);
+        g.drawString(lineNumber, x, y);
+
+        // Move to the next row
+
+        rowStartOffset = Utilities.getRowEnd(component, rowStartOffset) + 1;
+      } catch (Exception e) {
+        break;
+      }
+    }
+  }
+
+  /*
+   * We need to know if the caret is currently positioned on the line we
+   * are about to paint so the line number can be highlighted.
+   */
+  private boolean isCurrentLine(int rowStartOffset) {
+    int caretPosition = component.getCaretPosition();
+    Element root = component.getDocument().getDefaultRootElement();
+
+    if (root.getElementIndex(rowStartOffset) == root.getElementIndex(caretPosition))
+      return true;
+    else
+      return false;
+  }
+
+  /*
+   * Get the line number to be drawn. The empty string will be returned
+   * when a line of text has wrapped.
+   */
+  protected String getTextLineNumber(int rowStartOffset) {
+    Element root = component.getDocument().getDefaultRootElement();
+    int index = root.getElementIndex(rowStartOffset);
+    Element line = root.getElement(index);
+
+    if (line.getStartOffset() == rowStartOffset)
+      return String.valueOf(index + 1);
+    else
+      return "";
+  }
+
+  /*
+   * Determine the X offset to properly align the line number when drawn
+   */
+  private int getOffsetX(int availableWidth, int stringWidth) {
+    return (int) ((availableWidth - stringWidth) * digitAlignment);
+  }
+
+  /*
+   * Determine the Y offset for the current row
+   */
+  private int getOffsetY(int rowStartOffset, FontMetrics fontMetrics)
+    throws BadLocationException {
+    // Get the bounding rectangle of the row
+
+    Rectangle r = component.modelToView(rowStartOffset);
+    int lineHeight = fontMetrics.getHeight();
+    int y = r.y + r.height;
+    int descent = 0;
+
+    // The text needs to be positioned above the bottom of the bounding
+    // rectangle based on the descent of the font(s) contained on the row.
+
+    if (r.height == lineHeight) // default font is being used
+    {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this left curly brace to the end of previous line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ALeftCurlyBraceEndLineCheck)